kernel/src/main/java/org/kframework/parser/inner/generator/RuleGrammarGenerator.java

```diff
@@ -179,15 +187,19 @@ public static boolean isParserSort(Sort s) {
 
     /* use this overload if you don't need to profile rule parse times. */
     public static ParseInModule getCombinedGrammar(Module mod, boolean strict) {
-      return getCombinedGrammar(mod, strict, false, false, null);
+      return getCombinedGrammar(mod, strict, false, false, false, null);
     }
 
     public static ParseInModule getCombinedGrammar(Module mod, boolean strict, boolean timing, boolean isBison) {
-      return getCombinedGrammar(mod, strict, timing, isBison, null);
+      return getCombinedGrammar(mod, strict, timing, isBison, false, null);
     }
 
     public static ParseInModule getCombinedGrammar(Module mod, boolean strict, boolean timing, FileUtil files) {
-      return getCombinedGrammar(mod, strict, timing, false, files);
+      return getCombinedGrammar(mod, strict, timing, false, false, files);
+    }
+
+    public static ParseInModule getCombinedGrammar(Module mod, boolean strict, boolean timing, FileUtil files, boolean forGlobalScanner) {
+      return getCombinedGrammar(mod, strict, timing, false, forGlobalScanner, files);
     }
```

Comment:
Can we check which of these overloads are used directly and remove the ones that aren't (if there are any)? Maybe we don't need the one without `boolean forGlobalScanner` anymore?

Reply:
I'm confused about `forGlobalScanner` as well. Why is it needed?

Reply:
All of these overloads are used somewhere. In response to Radu's question, it's needed so that the scanner we generate to parse rules contains /all/ of the syntax and not just the syntax visible to the main module.

Reply:
If I understand correctly, you are trying to optimize the case when you are parsing `STDOUT-STREAM` and `STDIN-STREAM` by reducing the number of productions included in the scanner. IMO the possible gains in speed don't merit adding a new parameter here. So I would like either:
- remove the option. The benefits don't outweigh the added code complexity.
- add inline documentation explaining the functionality. I spent more than half an hour trying to understand why this is needed and how it works.

If I'm getting this wrong, please correct me.

Reply:
No see, if the main scanner for rules doesn't include private syntax, then we will need to generate a new scanner for every module containing any private syntax, which will be very expensive.

Reply:
I understand this. Why isn't this the default for every other scanner? Why do you need to make it an option?

Reply:
This is the only place where scanners are created that are used non-modularly for other modules. Every other place where a ParseInModule is being constructed, we either don't use the scanner that comes with that object, or else we use that scanner only to scan that module, in which case we don't /want/ to include the private syntax in the scanner.

Reply:
That sounds reasonable. Please add that explanation as a code comment. It's going to be useful in the future.